smartapps/dianoga/netatmo-connect.src/netatmo-connect.groovy

```diff
@@ -35,35 +38,53 @@ preferences {
 }
 
 mappings {
-	path("/receivedToken"){action: [POST: "receivedToken", GET: "receivedToken"]}
-	path("/receiveToken"){action: [POST: "receiveToken", GET: "receiveToken"]}
-    path("/auth"){action: [GET: "auth"]}
+	path("/oauth/initialize") {action: [GET: "oauthInitUrl"]}
+	path("/oauth/callback") {action: [GET: "callback"]}
 }
 
 def authPage() {
 	log.debug "In authPage"
-	if(canInstallLabs()) {
-		def description = null
 
-		if (state.vendorAccessToken == null) {
-			log.debug "About to create access token."
+	def description
+	def uninstallAllowed = false
+	def oauthTokenProvided = false
 
-			createAccessToken()
-			description = "Tap to enter Credentials."
+	if (!state.accessToken) {
+		log.debug "About to create access token."
+		state.accessToken = createAccessToken()
+	}
 
-			return dynamicPage(name: "Credentials", title: "Authorize Connection", nextPage:"listDevices", uninstall: true, install:false) {
-				section { href url:buildRedirectUrl("auth"), style:"embedded", required:false, title:"Connect to ${getVendorName()}:", description:description }
-			}
-		} else {
+	log.debug "STATE: ${state.authToken}"
```

Comment:
Might be a good idea to not log this in production since it pretty sensitive security info